recipes/ignition-math/all/conanfile.py

```diff
@@ -0,0 +1,143 @@
+import os
+
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class IgnitionMathConan(ConanFile):
+    name = "ignition-math"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://ignitionrobotics.org/libs/math"
+    description = " Math classes and functions for robot applications"
+    topics = ("ignition", "math", "robotics", "gazebo")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake", "cmake_find_package_multi", "pkg_config"
+    exports_sources = "CMakeLists.txt", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "16",
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _cmake_source_subfolder(self):
+        return "cmake_source_subfolder"
+
+    @property
+    def _cmake_build_subfolder(self):
+        return "cmake_build_subfolder"
+
+    @property
+    def _cmake_prefix_subfolder(self):
+        return "cmake_prefix_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        version_major = self.version.split(".")[0]
+        os.rename(
+            "ign-math-ignition-math{}_{}".format(version_major, self.version),
+            self._source_subfolder,
+        )
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn(
+                "{} recipe lacks information about the {} compiler support.".format(
+                    self.name, self.settings.compiler
+                )
+            )
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires c++17 support. The current compiler {} {} does not support it.".format(
+                        self.name,
+                        self.settings.compiler,
+                        self.settings.compiler.version,
+                    )
+                )
+
+    def requirements(self):
+        self.requires("eigen/3.3.7")
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/1.7.3")
```

Comment:
Why is it required?

Reply:
It was in the original PR 🤷‍♂️ it worked so I didn't ask 🙈

Reply:
Seems to be useless. eigen is the only dependency, and I would be surprised that underlying build files of ignition-math use pkg-config to consume eigen in a CMake based project.

Reply:
Good catch.. I'll test it locally since I have a few PR to follow up =)